src/eval.rs

```diff
@@ -921,6 +936,7 @@ fn apply_continuation<F: LurkField>(
                             store.intern_num(tmp)
                         }
                         Op2::Cons => store.cons(evaled_arg, *arg2),
+                        Op2::Begin => unreachable!(),
```

Comment:
why unreachable?

Reply:
oh, because handled before that?

Reply:
Because we don't need to do anything combining the results of evaluating the two arguments. So we can just return the second value. That means a `Binop2` continuation is never created. So I guess it's reachable by an illegal input that can't be constructed through normal evaluation.

I think for now, we haven't committed to creating proofs over such illegal input.

Reply:
so why make Begin an Op2?

Reply:
I wanted to avoid making a new continuation type, since that seemed simpler.

Reply:
But it does require handling two arguments, so we can't use Unop. It's kind of a hybrid 1.5Op.

Reply:
Also, I initially called it 'then', which I think is a better name than either begin or progn. I was going to implement begin in terms of then but realized this would leak a name and also wasn't necessary.

Reply:
@namin Is it going to be easy to implement PROG2 as a macro, or should we just handle it primitively? If so, maybe should just do it now.

Reply:
Sorry, I mean PROG1.

Reply:
I don't think we have variadic definitions at the user level. :(

Reply:
That's also going to be a problem for implementing `cond`.

Reply:
We should at *least* implement for macros. But as far as PROG1 goes, you could still manually write
```
(prog1 (emit 1)
  (begin
    (emit 2)
    (emit 3)))

=> 1
```

to sidestep the variadic question and focus on the sequencing.

Reply:
`(prog1 a b) => (let ((x a)) (begin b x))` so OK modulo hygiene

Reply:
The hygiene issue is not great. We need GENSYM, which will be a bit hard to reason about. Meanwhile, it's pretty easy to add BEGIN1 here and now, I guess (if that's what it should be called).

Reply:
I decided to just add. See updated description for more details.